framework/src/multiapps/MultiApp.C

```diff
@@ -502,6 +502,7 @@ MultiApp::createApp(unsigned int i, Real start_time)
   // Update the MultiApp level for the app that was just created
   app->setMultiAppLevel(_app.multiAppLevel() + 1);
   app->setupOptions();
+  preRunInputFile();
```

Comment:
@YaqiWang : no, it's part of the design that a sub app doesn't know that it's a sub app.  That is on purpose and is done for your own good ;-)

@snschune : ok.  This feels a little specific, but it's not a big deal.  I'm not against this.